NameGame/NameGame/APIService.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  APIService.swift
+//  NameGame
+//
+//  Created by Tammy Le on 6/12/17.
+//  Copyright © 2017 WillowTree Apps. All rights reserved.
+//
+
+import Foundation
+
+/**
+    Reads and models data from online API.
+ */
+
+class APIService {
+    
+    static let manager = APIService()
+    
+    // Url for user records API
+    let apiString: String = "https://willowtreeapps.com/api/v1.0/profiles/"
+    
+    // Load data from JSON file into a list of UserRecord objects.
+    func loadData(completion: @escaping ([UserRecord]) -> Void) {
+        
+        let url = URL(string: apiString)
+        let task = URLSession.shared.dataTask(with: url!) { data, response, error in
+            guard let data = data, error == nil, let response = response as? HTTPURLResponse, response.statusCode == 200 else {
+                print("Couldn't create shared session.")
```

Comment:
We need to make sure we call our closure with an empty array of users. If we fail to do so our app could end up in an unrecoverable state. Tomorrow we should also talk about an enum pattern we can use here that can help with further error handling called `Result`.